File: schemaorggo/mobileapplication.go
package schemaorggo

import "encoding/json"

// MobileApplication see : https://schema.org/MobileApplication
type MobileApplication struct {
	SoftwareApplication

	typeContext

	// CarrierRequirements see : https://schema.org/carrierRequirements
	// Specifies specific carrier(s) requirements for the application (e.g. an application may only work on a specific carrier network).
	// types : Text
	CarrierRequirements string `json:"carrierRequirements,omitempty"`
}

func (v MobileApplication) MarshalJSONWithTypeContext() ([]byte, error) {
	v.C = "http://schema.org"
	v.T = "MobileApplication"

	return json.Marshal(v)
}

func (v *MobileApplication) MarshalJSON() ([]byte, error) {
	if v == nil {
		return []byte("null"), nil
	}

	v.C = "http://schema.org"
	v.T = "MobileApplication"

	return json.Marshal(*v)
}

File: schemaorgphp/classes/mobile_application.php
<?php

class MobileApplication extends SoftwareApplication implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'MobileApplication';
	
	/**
	 * Specifies specific carrier(s) requirements for the application (e.g. an application may only work on a specific carrier network).
	 * see : https://schema.org/carrierRequirements
	 * @var string|string[]
	 */
	public var $carrier_requirements;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'MobileApplication'
		);
		
		$serialized = so_json_serialize( $this->carrier_requirements );
		if ( ! empty( $serialized ) ) {
			$out['carrierRequirements'] = $serialized;
		}
		
		return $out;
	}
}
